webservices/filters.py

```diff
@@ -25,23 +49,35 @@ def filter_range(query, kwargs, fields):
             query = query.filter(column <= kwargs[max_key])
     return query
 
+
 def filter_fulltext(query, kwargs, fields):
     for key, column in fields:
         if kwargs.get(key):
-            filters = [
-                column.match(utils.parse_fulltext(value))
-                for value in kwargs[key]
-            ]
-            query = query.filter(sa.or_(*filters))
+            exclude_list = [parse_exclude_arg(value) for value in kwargs[key] if is_exclude_arg(value)]
+            include_list = [value for value in kwargs[key] if not is_exclude_arg(value)]
+            if exclude_list:
```

Comment:
@ccostino and @LindsayYoung - the `filter_fulltext` changes are the part I feel could use the most scrutiny.

Reply:
I am mulling this over in my head and this all looks right to me.  The spot I'm not 100% on (but would be easy to check) is that the query gets constructed correctly given the mix of `and` and `or`.

I think this is constructed correctly, where things that are excluded are all `and` within the query, and things to be included are all combined with `or` (in other words:  a value is not `x` and `y` and `z`, but is `a` or `b` or `c`).  I just want to be sure. :-)

Reply:
Thanks @ccostino! I agree - when I checked the queries I was a bit surprised not to see parentheses in there. I updated the issue to do lots more of manual testing, especially for combo include/exclude for fulltext filters. I've tested a fair amount so far but I think I might take an inventory of which endpoints use `match`, `multi` and `fulltext` to see if I can do some more extensive/comprehensive manual tests.